### ISSUE TITLE
Suggestion/wish: combine some SAS code for both ADaM and SDTM as macros instead of two different pieces of code.

### ISSUE THREAD

**Issue by mhungria** [lexjansen/dataset-json-sas#16]
Suggestion/wish: combine some SAS code for both ADaM and SDTM as macros instead of two different pieces of code.
Refs: 02_create_metadata_from_definexml.sas and 05_compare_data.sas

**Comment by lexjansen** [comment]
For 05_compare_data.sas I can see this, as the parameters could be baselib and complib, without having to specify the dataset.  For 02_create_metadata_from_definexml.sas I'm not so sure, since the %CreateMetadataFromDefineXML macro already specifies the Define-XML file and the output library. The datastep after this macro call is really hardcoded. I'm not sure what macro signature you envision.

**Comment by mhungria** [comment]
i.e., %macro ManualUpdates(metadatalib=);

**Comment by lexjansen** [comment]
The code would be very specific for specific SDTM or ADaM study datasets, so I do not see how this could be generic.

**Comment by mhungria** [comment]
Since the main reason I separated the update of the keys into the 02_create_metadata_from_definexml.sas was to include it as a “manual adjustment”, I combined all manual adjustments as a macro and controlled it with the metadata lib indicated as a macro parameter. In case there may be other manual adjustments needed, having a unique place to adjust in the code is simpler. In any case, no need to follow the same if you don't foresee other manual adjustments.
